src/server/status_server/mod.rs

```diff
@@ -651,20 +651,20 @@ where
                             (Method::GET, "/status") => Ok(Response::default()),
                             (Method::GET, "/debug/pprof/heap_list") => {
                                 return Ok(make_response(
-                                    StatusCode::NOT_IMPLEMENTED,
-                                    "deprecated",
+                                    StatusCode::NOT_FOUND,
```

Comment:
IMO, it's better to use "METHOD_NOT_ALLOWED" or "GONE" to notify the caller that these methods are deprecated.